include/sys_matrix.h

```diff
@@ -0,0 +1,81 @@
+#ifndef SYS_MATRIX_H
+#define SYS_MATRIX_H
+
+#include "z64math.h"
+
+
+struct GraphicsContext;
+struct GameState;
+
+
+typedef enum {
+    MTXMODE_NEW,  // generates a new matrix
+    MTXMODE_APPLY // applies transformation to the current matrix
+} MatrixMode;
+
+
+typedef struct {
+    /* 0x00 */ u16 intPart[4][4];
+    /* 0x20 */ u16 fracPart[4][4];
+} MatrixInternal; // size = 0x40
+
+
+extern Mtx gIdentityMtx;
+extern MtxF gIdentityMtxF;
+
+
+/* Stack operations */
+void Matrix_Init(struct GameState* gameState);
+void Matrix_Push(void);
+void Matrix_Pop(void);
+void Matrix_Get(MtxF* dest);
+void Matrix_Put(MtxF* src);
+MtxF* Matrix_GetCurrent(void);
+
+/* Basic operations */
+void Matrix_Mult(MtxF* matrix, MatrixMode mode);
+void Matrix_Translate(f32 x, f32 y, f32 z, MatrixMode mode);
+void Matrix_Scale(f32 xScale, f32 yScale, f32 zScale, MatrixMode mode);
+void Matrix_RotateXS(s16 x, MatrixMode mode);
+void Matrix_RotateXF(f32 x, MatrixMode mode);
+void Matrix_RotateXFApply(f32 x);
+void Matrix_RotateXFNew(f32 x);
+void Matrix_RotateYS(s16 y, MatrixMode mode);
+void Matrix_RotateYF(f32 y, MatrixMode mode);
+void Matrix_RotateZS(s16 z, MatrixMode mode);
+void Matrix_RotateZF(f32 z, MatrixMode mode);
+
+/* Compound operations */
+void Matrix_RotateZYX(s16 x, s16 y, s16 z, MatrixMode mode);
+void Matrix_TranslateRotateZYX(Vec3f* pos, Vec3s* rot);
+void Matrix_RotateAndTranslateState(Vec3f* translation, Vec3s* rot);
```

Comment:
I'm not seeing this in the `c` file?

Reply:
Hmm, must have missed it when I removed some other old ones.